### DISCUSSION TITLE
Have ability have ovr toolkit or XSOverlay OSC to able type on VRCOSC as textbox for faster message

### DISCUSSION THREAD

**Discussion by NekoSuneVR** [VolcanicArts/VRCOSC#150] [Ideas]
somethign always think should be added, most of us have to type in chatbox but todo that we have to keep turning off program in VR able talk them in chatbox less way intergrate OVR toolkit and XSOverlay typing system OSC over it able message threw VRCOSC router mode

should something needs be done make peopel easy message people without turning off VRCOSC in VR

**Comment by VolcanicArts** [comment]
If you get the controls prefab you can disable VRCOSC sending to the ChatBox to let you manually type

**Reply by NekoSuneVR** [reply]
wheres about controls i need and would it be easyer have someting intergrated so they still have they clocks and etc

**Reply by VolcanicArts** [reply]
All the prefabs are always on the latest release

**Reply by NekoSuneVR** [reply]
something still have text ability type like have speech to text but using OVR Toolkit / XSOverlay chat typing OSC

**Reply by VolcanicArts** [reply]
OVRToolkit has a typing mode for the ChatBox and should still work without using the router

**Reply by NekoSuneVR** [reply]
it overrides it on VRCOSC so makes it rate limit alot have 2 OSC in 1 when could have ability have threw VRCOSC keep rate limit minimal

**Reply by NekoSuneVR** [reply]
everything i have right now already using router mode on VRCOSC right now part OVR toolkit typing system

**Reply by VolcanicArts** [reply]
The router doesn't impose a rate limit to stop VRC from complaining. The rate limit only happens if you send too much in a small period of time. If it's only happening during lag spikes then that's natural

**Reply by NekoSuneVR** [reply]
thinkign somethign coudl do for it but idk what can do 
![image](https://github.com/VolcanicArts/VRCOSC/assets/18028479/ff004153-0684-4192-bde5-984ada6fd376)
this example i would see if can do for us make things easyer

that way can put that as top primary show

**Reply by NekoSuneVR** [reply]
if look my setup, setup already done but cant always type in direct OVR screen when OVR has OSC make things easyer 
![image](https://github.com/VolcanicArts/VRCOSC/assets/18028479/acec2eba-be43-4148-b287-24cfd7500f9a)

**Reply by VolcanicArts** [reply]
Having variables for OVRToolkit and XSOverlay aren't possible as they write directly to VRChat's ChatBox endpoint so I can't access that data

**Reply by NekoSuneVR** [reply]
hmm is there any other way make it support VR typing make easy

**Reply by VolcanicArts** [reply]
The only thing I can offer is the controls prefab to stop it sending to the ChatBox. VRC's ChatBox is primitive since you just send it what words to display and that's it. If multiple apps try to send data to it at once then it just displays whatever sent most recently or times out if you send too much.

I was working on a physical keyboard for your avatar at one point which I can probably dust off with some of the recent changes I've made since it will be a bit easier to make now.

**Reply by NekoSuneVR** [reply]
if is going be useful send normal text then keep me updated for testing.
